### DISCUSSION TITLE
Exact search in osu!Lazer

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#24005] [Ideas] [answered]
Can you add a feature where you can find find exact matches in song select? I have just over 33000 maps, and if I'm looking for a specific song, it's quite challenging. :(

**Comment by peppy** [comment]
Could you give an example of a search term and screenshots showing the matches you're getting from it?

**Reply by ghost** [reply]
I'll send some screenshots later, but here's an example: I really like this song called "you". By simply searching "you", it narrows it down from ~33000 maps to ~4000. This search term also includes songs with artists, difficulty names, etc.. By typing title=you, I can get down to ~1100 results. This search term includes titles that have the word "you" (this should be pretty self explanatory). If I type "you m1v", I find the exact result (the artist's name is m1v), but this is inconvenient because it assumes I know the song's artist. (I had to open osu! and find the song to find it) What I'm suggesting is a way to only see songs with the EXACT search term. So if I search something like "exact=you" it only shows songs with the title/artist/mapper "you". All the tests are done on osu!stable because I have a LOT of maps and it would take a long time to import to lazer. If you want me to conduct this test on osu!lazer, I can, but it will take a little while. If this were implemented to osu!lazer, I think it would be better for it to be a toggle button beside the search than one of those unintuitive prefix things. Thanks for reading, like I said, I'll send a few screenshots later.

**Reply by ghost** [reply]
*screenshots taken in stable*

1: https://imgur.com/a/vh5FIzN
2: https://imgur.com/a/bywYJNN
3: https://imgur.com/a/IftA48H
4 photoshopped hypothetical (how I want it to work): https://imgur.com/a/IUMDlbf

Obviously I want this in both versions, but lazer is probably more important.

**Comment by monochrome22** [comment]
Maybe add being able to use quotation marks as a search option to search for exact matches, `"you"`.

**Reply by peppy** [reply]
This is the only valid way to implement this feature, correct.

**Reply by ghost** [reply]
oh I forgot quotation marks are pretty much standard when it comes to specific searches lmao

**Comment by peppy** [comment, marked as answer]
#24045

**Reply by ghost** [reply]
will this also be implemented in osu!web/osu!direct/whatever the lazer equivalent is called?

**Reply by peppy** [reply]
those already support the same implementation

**Reply by ghost** [reply]
cool 👍

**Reply by monochrome22** [reply]
Just to let you know, that implementation wouldn't help the OP much.
With 70k beatmaps:
![you](https://github.com/ppy/osu/assets/122969285/03a86fa6-a9ea-426f-bf39-89278b86ca8b)
(`title=` doesn't work in lazer)

I know in the PR thread it was mentioned that exact match would be added later, I wanted clarify its importance.
Double quotes or `'` might work for exact match.
Or maybe just adding `exact=` would be fine, because it would probably narrow down the search enough with any keyword, so using it with `title=` or `artist=` wouldn't be necessary.

**Reply by peppy** [reply]
title= *does* work, as tested in the PR

I'm thinking `@""` may work well for exact-full-phrase match mode

ie.

`@"you"`
or
`title=@"you"`

Another option would be `"you"!`, which has the benefit of being able to append it to an existing search (and quickly remove it too).

@bdach curious on your thoughts here.

**Reply by monochrome22** [reply]
Do you mean there are other changes that will make title= work? Because it never worked for me, not with the PR and not in normal client.
![artist](https://github.com/ppy/osu/assets/122969285/7e92b50a-cb22-48d3-be6d-84c0fc22c4dd)
![title](https://github.com/ppy/osu/assets/122969285/84c52546-ab66-4f67-b93b-ec925d7b3a20)

**Reply by peppy** [reply]
> Do you mean there are other changes that will make title= work? Because it never worked for me, not with the PR and not in normal client.

Fixed in #24057.

> Another option would be `"you"!`, which has the benefit of being able to append it to an existing search (and quickly remove it too).

Added in #24058
